lib.php

```diff
@@ -3628,7 +3628,7 @@ function facetoface_print_session($session, $showcapacity, $calendaroutput=false
     // Display requires approval notification.
     $facetoface = $DB->get_record('facetoface', array('id' => $session->facetoface));
 
-    if ($facetoface->approvalreqd) {
+    if (get_config(null, 'facetoface_enableapprovals') && $facetoface->approvalreqd) {
```

Comment:
Could we move this to it's own function. The check appears to be called in multiple places and is especially confusing when used in the negative (e.g. not config OR not approval reqd)

Alternatively, the approvalreqd property could be dissected and updated to only be true if the config is set, and then should already be correct in other locations.

How does that sound?

Reply:
The `approvalreqd` setting is per activity instance. It should be not be reset based on the plugin setting `enableapprovals`. If activities are set up to use approvals, and then an unwitting site admin disabled, then re-enabled the `enableapprovals` setting the activity settings should be in the same state as before.

This pattern of check only appears in two places which is why I didn't turn it into a function (Normally use rule of 3), but I can do to make it more readable.

Reply:
Added new helper method.

Reply:
> This pattern of check only appears in two places which is why I didn't turn it into a function (Normally use rule of 3), but I can do to make it more readable.

Thanks Andrew, I noticed 3 places it is used, namely, the two places the helper function was used instead:
```
function facetoface_manager_needed($facetoface) {
    return get_config('facetoface', 'enableapprovals')
        && (
            $facetoface->approvalreqd
            || $facetoface->confirmationinstrmngr
            || $facetoface->reminderinstrmngr
            || $facetoface->cancellationinstrmngr
        );
}
```
Previously the logic only applied the inner conditional. And the new check whilst it makes sense, might also take the following form, which should still technically prove true, and be a bit shorter - reasoning being that the approval required goes hand in hand with the helper function.

```
function facetoface_manager_needed($facetoface) {
    return \mod_facetoface\helper::is_approval_required((object) $facetoface)
        && (
            $facetoface->confirmationinstrmngr
            || $facetoface->reminderinstrmngr
            || $facetoface->cancellationinstrmngr
        );
}
if () {
```

Reply:
Hi @keevan , I would have simplified, except the logic is not the same for checking whether approvals are required versus is a manager is needed.

There are two scenarios where the above are not equal (I can provide all logic pairs if you want too).

For simplication: a = get_config('facetoface', 'enableapprovals'), b = $facetoface->approvalreqd, c = $facetoface->confirmationinstrmngr || $facetoface->reminderinstrmngr || $facetoface->cancellationinstrmngr

1. a = true, b = false, c = true
  Existing logic: true, New logic: false
2. a = true, b = true, c = false
  Existing logic: true, New logic: false
  
These two methods should remain decoupled as they are checking different states. I can add in unit tests for the existing function `facetoface_manager_needed` if you think it's required?

Reply:
Ah, yup you're right.

The odd bit I find from reading this, is the function returns if the manager is needed, but it's dependent on all those factors. Confirmation/Reminder/Cancellation all seem like email notifications, which only works if manager approval is enabled per the new string change.

> $string['manageremaildisabled'] = 'Manager\'s email settings are not available if manager approvals are not enabled.';

Side note: the language between manager approvals being enabled sitewide, vs required for the activity, is subtle and can be confusing.

Scenario 1: 
**A manager is needed** if managers can approve sitewide, approval is not required, and some notification is enabled

It's an odd case but I'm sure it happens. I'm okay with keeping your patch as you've explained the states.